python/tests/variant_tests/dtype_and_variant.py

```diff
@@ -0,0 +1,369 @@
+from tests.aggregate_tests.aggtst_base import TstView, TstTable
+from decimal import Decimal
+
+
+# INTEGER
+class varnttst_int_tbl(TstTable):
+    """Define the table used by the integer tests"""
+
+    def __init__(self):
+        self.sql = """CREATE TABLE varnt_int_tbl(
+                      tiny_int TINYINT,
+                      small_int SMALLINT,
+                      int INT,
+                      bigint BIGINT)"""
+        self.data = [{"tiny_int": 20, "small_int": 24, "int": 21, "bigint": 23}]
+
+
+class varnttst_int_to_variant(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [
+            {
+                "tiny_int_varnt": "20",
+                "small_int_varnt": "24",
+                "int_varnt": "21",
+                "bigint_varnt": "23",
+            }
+        ]
+        self.sql = """CREATE MATERIALIZED VIEW int_to_variant AS SELECT
+                      CAST(tiny_int AS VARIANT) AS tiny_int_varnt,
+                      CAST(small_int AS VARIANT) AS small_int_varnt,
+                      CAST(int AS VARIANT) AS int_varnt,
+                      CAST(bigint AS VARIANT) AS bigint_varnt
+                      FROM varnt_int_tbl"""
+
+
+class varnttst_variant_to_int(TstView):
```

Comment:
can you also add tests where the types do not match?
we should be able to cast int -> variant -> bigint, and even to tinyint if the value is small enough.

Reply:
this can be in a separate PR too